### ISSUE TITLE
Implement automatic chunked-writing of variables if they exceed a certain size threshold

### ISSUE THREAD

**Issue by cisaacstern** [pangeo-forge/pangeo-forge-recipes#164]

**Comment by cisaacstern** [comment]
This issue is a blocker for running the interior recipes of https://github.com/pangeo-forge/staged-recipes/pull/24, which contain four single-variable arrays of shape `(30, 107, 735, 607)` (for dims `'time'`, `'deptht'`, `'y'`, `'x'`, respectively) and size ~6 GBs.

As observed by @TomAugspurger in https://github.com/pangeo-forge/pangeo-forge-recipes/issues/93#issuecomment-841824724, these arrays crash execution at:

https://github.com/pangeo-forge/pangeo-forge-recipes/blob/e1ef5756ced85e5f9b029e1c40291e133a73329b/pangeo_forge_recipes/recipes/xarray_zarr.py#L496-L498

Tom, in the same above-linked comment, you hypothesized

>  So if users have set Recipe.target_chunks then just doing ds.chunk(self.target_chunks) should be fine. I'll test it out on this dataset.

Did you ever get started on this? If not, I'll start a draft PR which integrates this insight with the chunked writing structure outlined by @rabernat in https://github.com/pangeo-forge/pangeo-forge-recipes/issues/93#issue-850874898. So something like:

```python
data = ds_chunk[varname].chunk(self.target_chunks).data
with dask.config.set(scheduler="single-threaded"): 
    dask.array.store(data, zarr_array, lock=false, regions=zarr_region)
```

**Comment by TomAugspurger** [comment]
I have a start at https://github.com/TomAugspurger/pangeo-forge/tree/feature/large-datasets, but haven't had a chance to finish it off. There will be merge conflicts with `master` now that the serialization PR is in.

**Comment by cisaacstern** [comment]
Closed by #166
